### PR TITLE
Fix E2E on Windows

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Enable KeepAlive for E2E http client.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: endgame

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/e2e/utils/helpers.go [hunk 1]
```diff
@@ -81,7 +81,8 @@ func newHTTPClient() http.Client {
 			Transport: &http.Transport{
```
Comment (pkedy) [inline tests/e2e/utils/helpers.go:81]: We probably want to specify a timeout on the `http.Client` too.

```go
Timeout: time.Second * 15,
```

Comment (artursouza) [reply tests/e2e/utils/helpers.go:81]: Can this be in another PR? Rerunning this takes 1h and we need to cut a new RC.

**Review by wcs1only — APPROVED** [review]

File: tests/e2e/utils/helpers.go (hunk 1 above)
Comment (pkedy) [reply tests/e2e/utils/helpers.go:81]: 💯

**Review by pkedy — APPROVED** [review]

**Pull request merged**